demo/init-container-failure.demo

```diff
@@ -0,0 +1,25 @@
+# Demo: inject fault to initContainer
```

Comment:
Why use `kwok cluster` and `existing cluster` to run the same demo?

Reply:
Maybe we can combine these two like this

``` bash
kwokctl create cluster --enable-crds Stage
kubectl apply -f https://github.com/kubernetes-sigs/kwok/releases/download/v0.5.2/stage-fast.yaml
kubectl apply -f https://github.com/kubernetes-sigs/kwok/raw/main/kustomize/stage/pod/chaos/pod-init-container-running-failed.yaml
```

With `--enable-crds Stage` the kwok cluster will fetch the stage from the API server.

Reply:
> Why use `kwok cluster` and `existing cluster` to run the same demo?

Demonstrate different ways to run the demo. In our use cases where we run additional controllers/operators, we use kwok in cluser with real control plane nodes and virtual nodes.